---
title: "A Chemokine Signature of T Cell Exhaustion and Metabolic Reprogramming in Renal Cell Carcinoma"
subtitle: "Methods, Results, Figures and Tables"
author: "Uro study team"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::word_document2:   
    reference_docx: ms_template.docx
    
bibliography: cxcl_biblio.bib
csl: frontiers-in-immunology.csl
header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}}       \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
---

```{r, setup, include = FALSE}

knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, 
                      dev = "png", dpi = 600, 
                      tab.cap.pre = 'Supplementary Table ', tab.cap.sep = " ", 
                      tab.cap.style = 'Table Heading')

set_flextable_defaults(font.family = 'Cambria', 
                       font.size = 10)


```

\newpage

# Abstract

__Background:__ 
Molecular and immunological determinants of survival and treatment response in renal cell cancer (RCC) are poorly characterized. 
Intriguingly, the key mediators of anti-tumor immunity and anti-immune checkpoint (IC) therapy, cytotoxic CD8^+^ T cells appear to be dysfunctional and not to confer survival and therapeutic benefit in RCC. 
Herein, we delved into molecular pathways of chemokine-driven T cell recruitment and suppression in the microenvironment of RCC.

__Methods:__
Eight publicly available bulk RCC transcriptome collectives and one single cell RNAseq dataset were analyzed. 
xCell and QuanTIseq immunodeconvolution, correlation analysis, semi-supervised clustering, Cox modeling of survival, gene set variation analysis and Monte Carlo-based modeling of metabolic reaction activity were employed to elucidate chemokine-driven recruitment mechanisms of CD8^+^ T lymphocyte recruitment as well as immune, molecular and metabolic processes leading to their functional anergy in RCC.

__Results:__ 
We found the chemokine genes _CXCL9/10/11_, _CXCL13_, _XCL1_ and genes of their receptor genes _CXCR3_, _CXCR5_ and _XCR1_ to be significantly enriched in RCC as compared with the normal kidney and to be strongly associated with tumor-infiltrating effector memory and central memory CD8^+^ T cells. 
Tumor associated macrophages, activated myeloid dendritic cells and, to a lesser degree, malignant cells were identified as primary sources of _CXCL9/10/11_, _CXCL13_ and _XCL1_. 
Expression of their cognate receptors was highly specific for CD8^+^ T cells, as revealed by single cell RNAseq. 
Yet, none of the investigated genes could be equivocally associated with significantly altered mortality, relapse and anti-IC therapy response in multiple analyzed cohorts. 
The cluster of cancer samples characterized by high chemokine expression and CD8^+^ T cell infiltration displayed a strong activation of interferon/JAK/STAT signaling, elevated expression of multiple T cell exhaustion-associated transcripts. 
The chemokine high RCC subset was also hallmarked by suppression of oxidative energy metabolism and predicted high activity of IDO1-mediated tryptophan catabolism to immunosuppressive kynurenin and quinolinic acid.

__Conclusion:__
We propose a _CXCL9/10/11_-, _CXCL13_- and _XCL1_-mediated mechanism of CD8^+^ T cell recruitment. 
The anti-tumor potential of those cytotoxic T lymphocytes remains constrained by widespread IC activity in the RCC microenvironment, metabolic re-programming and IDO-mediated immunosuppression. 
A therapy regimen targeting IC circuits, energy metabolism of RCC and IDO activity may pose an attractive treatment option to unleash anti-tumor T cell immunity.

\newpage

# Methods

Detailed description of analysis procedures is provided in __Supplementary Methods__.

## Ethics

No ethics committee approval was needed for publicly available anonymous expression and clinical information data sets. 
Donors of tissue for flow cytometry measurements provided written consent. 

## Software

The analysis was done with R version 4.2.0. 
Basic data transformation tasks were accomplished with the _tidyverse_ package bundle [@Wickham2019] and the development package _trafo_ (https://github.com/PiotrTymoszuk/trafo). 
Exploratory data analysis and hypothesis testing was done with the _rstatix_ [@Kassambara2021], _ExDA_ (https://github.com/PiotrTymoszuk/ExDA) and _microViz_ (https://github.com/PiotrTymoszuk/microViz) packages. 
Network analysis was accomplished with _igraph_ [@Csardi2006]. 

Results were visualized with the packages _ggplot2_ (bubble plots) [@Wickham2016], _ExDA_ (box plots), _microViz_ (Forest plots, Volcano plots, bar plots of regulation estimates and p values), _survminer_ (Kaplan-Meier plots) [@Kassambara2016] and _ggnetwork_ (network plots) [@Briatte2021]. 

## Data import and transformation

The following publicly available RCC data sets were re-analyzed: 
TCGA Clear Cell Carcinoma (KIRC) project [@Creighton2013], 
whole-genome subsets of the CheckMate 010 (CM 010) and CheckMate 025 (CM 025) studies [@Braun2020], 
GSE73731 [@Wei2017], 
GSE167093 [@Laskar2021], 
RECA-EU [@RECA] 
and E-MTAB 1980 [@Sato2013]. 
Author-provided expression data and clinical information for the TCGA cohort were extracted from the GDC Data Portal with the TCGA-Assembler script (https://github.com/compgenome365/TCGA-Assembler-2/blob/master/TCGA-Assembler/). 
Author-provided expression and clinical data for GSE73731 and GSE167093 were fetched with the _GEOquery_ package [@Sean2007]. 
E-MTAB 1980 and RECA-EU data sets were imported from the ArrayExpress and ICGC Data Portal repositories, respectively, with in-house developed R scripts.
To investigate possible effects of therapy with immune checkpoint inhibitors, the CheckMate 025 everolimus (CM 025 EVER) and nivolumab (CM 025 NIVO) treatment arms were analyzed separately.

For the microarray expression studies (GSE73731, GSE167093, E-MTAB 1980), integration of multiple probes targeting the same gene was accomplished by geometric mean. 
Expression values were transformed with $log_2(transcript count + 1)$ (RNA sequencing: TCGA, CM 010, CM 025, RECA-EU) of $log_2$.
Immune infiltration estimates were calculated using the QuanTIseq and xCell algorithms (_immunedeconv_ package) [@Finotello2019; @Sturm2020; @Aran2017].

Gene signatures corresponding to Reactome pathways were extracted from the MSig database, version 7.5.1, and the signatures' single sample gene set enrichment analysis scores (ssGSEA) were calculated with the _GSVA_ algorithm [@Hanzelmann2013]. 
Genes associated with T cell exhaustion were retrieved from four recent papers [@Cai2020; @Woroniecka2018; @Kocher2022; @Zhang2022].

## Genes of interest

The chemokine genes of interest were identified within the set of `r length(screen$variables)` chemokine genes available for all investigated datasets (_`r paste(screen$variables, collapse = ', ')`_). 
The chemokine genes were screened in the TCGA KIRC cohort for differences in expression levels between the RCC and normal kidney tissue (paired normal/tumor samples, two-tailed paired T test, function `test_two_groups()`, _microViz_ package), and for correlation with CD8^+^ T cell content in RCC predicted with the QuanTIseq algorithm [@Finotello2019] (Spearman's correlation, `correlate_variables()`, package _ExDA_). 

Among `r length(screen$variables)` chemokine genes measured in all investigated collectives, expression of 
_CXCL9_, _CXCL10_, _CXCL11_, _CXCL13_ and _XCL1_ was found to be highly enriched in the malignant tissue (paired two-tailed T test) and strongly significantly associated with CD8^+^ T cell levels (Spearman's correlation) in the TCGA KIRC cohort (__Supplementary Figure S1__). 
These chemokine genes and their cognate receptors _CXCR3_ (receptor for CXCL9/10/11), _CXCR5_ (CXCL13) and _XCR1_ (XCL1) were investigated further in the current report.

## Comparison of gene expression between the normal and tumor samples, correlation with non-malignant cell infiltration

log~2~-transformed expression was compared between the donor-matched normal kidney and cancer samples by two-tailed paired T test and Cohen's d effect size statistic. 
Correlation of log~2~-transformed gene expression with QuantIseq- and xCell-predicted non-malignant cell content [@Finotello2019; @Aran2017] was investigated with Spearman's test. 
The expression comparison and correlation results were corrected for multiple testing with the false discovery rate (FDR) method [@Benjamini1995]. 
The scaled Spearman's correlation matrices for the genes of interest and cell types associated with them with moderate-to-large strength (correlation coefficient $\rho$ > 0.4) were visualized as force-directed network plots (packages _igraph_ and _ggnetwork_) [@Csardi2006; @Briatte2021].

## Survival modeling

Correlation of log~2~-transformed gene expression with overall or relapse-free survival by uni-variable Cox proportional hazard modeling [@Therneau2000] including the linear and spline term for the gene expression variable. 
Significance for model terms was corrected with FDR for multiple testing. 
No significant associations of the gene expression splines with overall or relapse-free survival could be observed (not shown).
Linear term model estimates are presented as hazard ratios (HR) with 95% confidence intervals.

Multi-parameter modeling of overall survival was done by Ridge regularized Cox regression (package _glmnet_) [@Friedman2010]. 
The explanatory factors included age, squared age, sex, tumor grade, tumor stage, metastasis stage) variables and log~2~-transformed expression values of the genes of interest. 
Two models were trained in the TCGA KIRC cohort: a model with demographic/grade/stage variables only and a model with the complete set of explanatory factors and both models were subsequently validated in the E-MTAB 1980 and RECA-EU collectives. 
To assess the add-on effect of gene expression on the survival prediction accuracy as compared with the demographic/grade/stage information, concordance indexes of the full models and the demographic/grade/stage-only models were compared.

## Semi-supervised clustering

By PAM (partition around medoids) - cosine distance clustering [@Schubert2019; @Drost2018] in respect to to normalized log~2~-transformed expression of the genes of interest, two clusters of RCC samples, the chemokine high and chemokine low cluster were defined in the training TCGA KIRC cohort. 
The clustering algorithm was chosen based on its excellent reproducibility in 10-fold cross validation [@Lange2004] and good explanatory performance as compared with hierarchical and KMEANS clustering algorithms (__Supplementary Figure S7A__). 
Choice of the cluster number (k = 2) was motivated by the bend of the within-cluster sum of squares curve, the peak mean silhouette statistic and visual inspection of the heat map of distances between cancer samples (__Supplementary Figure S7BC__).

Prediction of the chemokine cluster assignment in the test collectives (E-MTAB 1980, GSE73731, GSE167093, RECA-EU and CheckMate cohorts) was done with a 15-nearest neighbor (15-NN) classifier with inverse distance weighting. 
The prediction yielded clustering structures with comparable fractions of explained variance and comparable cluster sizes in the training and the test cohorts indicative of high reproducibility (__Supplementary Figure S8A__).

xCell estimates of non-malignant cell infiltration [@Aran2017] were compared between the chemokine clusters by FDR-corrected Mann-Whitney U test. 
Differences in distribution of tumor stages, MSKCC risk strata and frequencies of best overall therapy response (complete/partial response versus stable/progressive disease) between the clusters were compared by FDR-corrected $\chi^2$ test. 
Differences in overall and relapse-free survival between the clusters were assessed by Peto-Peto test [@Kassambara2016].

## Differences in Reactome pathways, gene expression and signaling modulation between the clusters

Differences in ssGSEA scores [@Hanzelmann2013] of Reactome pathway gene signatures between the chemokine clusters were investigated by FDR-adjusted two-tailed T test (__Supplementary Table S1__). 
Genes differentially expressed between the chemokine clusters were identified by FDR-corrected two-tailed T test and > 1.25 fold-regulation in the chemokine high versus chemokine low cluster (__Supplementary Tables S2__ - __S3__). 
Modulation of KEGG-listed signaling pathways in chemokine high versus chemokine low cluster cancers based on the differential gene expression was investigated with _SPIA_ [@Tarca2009] (__Supplementary Table S4_).

## Biochemical reaction modulation in the CXCL9 expression strata

Rules of assignment of genes to biochemical reactions were retrieved from the Recon2 human metabolism model available via the BiGG database [@King2016]. 
Regulation of biochemical reactions between the chemokine high and chemokine low clusters based on whole-genome estimates of differential gene expression and their standard errors was done by evaluation of the gene assignment rules with the 'gene - protein - reaction' (GPR) principle [@Gavai2015].
Standard deviation, 95% confidence intervals and p values for the predicted reaction regulation estimates were obtained by a Monte Carlo simulation (n = 1000 draws) (__Supplementary Table S5__). 
The analysis was done with the package _BiGGR_ [@Gavai2015] and the development package _biggrExtra_ (https://github.com/PiotrTymoszuk/biggrExtra). 

## Data and code availability

Data analyzed within the study are publicly available. 
The study pipeline is available at https://github.com/PiotrTymoszuk/lymphokine_RCC.

\newpage

# Results 

## Characteristic of the study cohorts

Altogether, we analyzed clinical and transcriptome data from 8 publicly available RCC cohorts including a total of `r clust_dge$analysis_tbl %>% map_dbl(nrow) %>% sum` cancer samples. 
Males constituted between 57 and 76% of investigated patients and the gender distribution was comparable between the cohorts. 
The median age ranged between 60 and 64 years and was similar in the study collectives. 
The RECA-EU, GSE167093 and E-MTAB 1980 cohort individuals tended towards lower tumor grades (G2) as compared with the remaining collectives with the grade data available. 
The majority of tumors was stage T1. 
In the TCGA, RECA-EU and E-MTAB 1980, information on initial metastasis status was available; only a small fraction of participants had lymph node or distant metastases at RCC diagnosis. 
The median observation time was noticeably shorter in the CheckMate (CM) 010 and CM 025 collectives. 
In these cohorts, the rate of relapses and mortality within the observation time window was substantially higher. 
In the samples from the CheckMate ICI therapy studies, between 5.1 (CM 025 everolimus) and 25% patients (CM 010 and CM 025, nivolumab) displayed an overall therapy response defined as complete or partial response (__Table \@ref(tab:tab-cohort)__).

## *CXCL9/10/11*, *CXCL13* and *XCL1* are key chemokines for cytotoxic T cell recruitment in RCC

First, we sought to identify RCC-specific chemokines which may mediate CD8^+^ T cells recruitment. 
To this end, expression of `r length(screen$variables)` chemokine genes was compared between RCC and the normal kidney, and correlated with CD8^+^ T cell infiltration predicted by the QuanTIseq algorithm [@Finotello2019] in the TCGA KIRC cohort. 
In such analysis, _CXCL9_, _CXCL10_, _CXCL11_, _CXCL13_ and _XCL5_ were found to be at lest four-fold upregulated in the cancer tissue and correlated strongly with the predicted CD8^+^ lymphocyte infiltration with $\rho$ > 0.6 in Spearman test (__Supplementary Figure S1__). 
These chemokine genes along with genes coding for their cognate receptors: _CXCR3_ (receptor for CXCL9/10/11), _CXCR5_ (receptor for CXCL13) and _XCR1_ (XCL1) were analyzed further in more detail.

_CXCL9_, _CXCL10_, _CXCL13_ and _XCL1_ chemokine transcripts and the _CXCR3_, _CXCR5_ receptor transcripts were significantly enriched in RCC as compared with the non-malignant tissue in an analysis of donor-matched samples in the TCGA KIRC and GSE167093. 
This upregulation was particularly strong for _CXCL9_, _CXCL10_ and _CXCL13_. 
In turn, significantly elevated expression of _CXCL11_ and _XCR1_ in RCC as compared with the non-malignant kidney could be observed only in the TCGA KIRC cohort but not in the GSE167093 collective (__Figure \@ref(fig:fig-norm-tumor)__). 

As revealed by network analysis of Spearman's correlations between the gene expression and non-malignant cell content predicted by the xCell algorithm [@Aran2017], expression of _CXCL9/10/11_, _CXCR3_, _CXCL13_, _XCL1_ and _XCR1_ was tightly associated with increased infiltration by pan-CD8^+^, central memory and effector memory CD8^+^ T cells, activated myeloid and plasmacytoid dendritic cells (mDC and pDC), tumor-associated macrophages (TAM) and , CD4^+^ T lymphocytes and B cells (__Figure \@ref(fig:fig-xcell)__, __Supplementary Figure S2__). 
This finding could be corroborated by a correlation analysis in the majority of the investigated cohorts. 
The association of CD8^+^ T cell levels with _CXCL9_, _CXCL10_ and _CXCL13_ cancer tissue expression was the strongest. 
Substantially weaker correlations were detected for cytotoxic T cells and _CXCR5_ or _XCR5_ (__Supplementary Figures S3 - S4__). 
Analogical, highly reproducible correlation could be observed between RCC expression of the genes of interest and CD8^+^ T cell infiltration levels as predicted by the QuanTIseq algorithm [@Finotello2019] (__Supplementary Figure S5 - S6__). 
Interestingly, levels of neither of _CXCL9/10/11_, _CXCR3_, _CXCL13_, _XCL1_ or _XCR1_ could be significantly and reproducibly associated with regulatory T cell (Treg) infiltration.

-- Place for the scRNAseq results --

PAM clustering of the TCGA KIRC samples in respect to normalized expression levels of _CXCL9/10/11_, _CXCR3_, _CXCL13_, _XCL1_ and _XCR1_ yielded two distinct subsets of RCC termed further the chemokine high and chemokine low clusters (__Supplementary Figure S7__). 
This clustering structures was highly conserved in the remaining investigated collectives as evident from comparable fractions of explained clustering variance and similar distribution of chemokine high and chemokine low sample frequencies between the cohorts (__Supplementary Figure S8A__). 
The strongest differences in expression of the clustering genes between the chemokine high and low cluster cancers were detected for _CXCL9/10/11_, _CXCR3_ and _CXCL13_ (__Figure \@ref(fig:fig-lympho-clust)__, __Supplementary Figure S8B__). 
In line with the correlation results, xCell-predicted infiltration of central memory CD8^+^ T cells was significantly higher in the chemokine high than in the chemokine low cluster in all investigated collectives. 
Levels of xCell-predicted TAM, activated mDC RCC and B cell infiltration were significantly upregulated in chemokine high cancers in the majority of analyzed cohorts (__Figure \@ref(fig:fig-clust-infil)__, __Supplementary Figure S9__).

Collectively, the results of correlation, single cell RNA sequencing and semi-supervised clustering put forward _CXCL9/10/11_, _CXCL13_ and _XCL1_ chemokine genes along with their receptors _CXCR3_, _CXCR5_ and _XCR1_ as key molecules involved in recruitment of effector memory and central memory CD8^+^ T cells to the RCC. 
TAM and activated mDC and, to a lesser extent, malignant cells are put forward as the major sources of the investigated chemokines in the RCC microenvironment.

## No effects of *CXCL9/10/11*, *CXCL13* and *XCL1* on survival and immune checkpoint therapy response

Except for the TCGA KIRC cohort displaying a significant enrichment of late stage tumors in the chemokine high cluster, no significant differences in distribution of tumor stages or MSKCC risk groups could be observed between the clusters (__Supplementary Figure S10__). 
No differences in best therapy response between the chemokine clusters could be detected either for everolimus treatment or for the ICI therapy with nivolumab in the CheckMate collectives (__Supplementary Figure S11__). 
Similarly, overall and relapse-free survival times in the clusters did not differ significantly (__Figure \@ref(fig:fig-clust-os)__, __Supplementary Figure S12__).

At the level of single genes, an approximately 30% increase in overall mortality risk could be found in the CheckMate 025 nivolumab cohort could be associated with _CXCL10_ and _CXCR3_ expression as revealed by uni-variable Cox modeling. 
For _CXCR3_, significantly shorter survival was also observed in the TCGA KIRC collective. 
Intriguingly, high RCC expression was linked to a significantly better overall survival in the RECA-EU collective (__Supplementary Figure S13__). 
High expression of _CXCL13_ and its cognate receptor _CXCR3_ correlated with significantly shorter overall survival in the CheckMate 025 nivolumab and TCGA KIRC collectives. 
Analogically, _XCL1_ was linked to a higher death risk in the TCGA KIRC cohort. 
Its receptor _XCR1_ was associated with significantly better overall survival in the TCGA KIRC and RECE-EU collectives and was linked to poor overall survival in the E-MTAB 1980 and CheckMate nivolumab cohorts (__Supplementary Figure S14__).
Significantly shorted relapse-free survival was observed for high RCC levels of _CXCL13_, _CXCR5_ and _XCL1_ in the TCGA KIRC dataset, _XCR1_ was in turn a good prognostic marker in the same collective. 
However, these observations could not be reproduced in the remaining investigated cohorts. 
No significant effects of _CXCL9/10/11_ or _CXCR3_ on relapse risk could be discerned (__Supplementary Figures S15 - S16__).

Finally, in multi-parameter Ridge Cox modeling of overall survival as a function of age, sex, grade, tumor and metastasis stage together with gene expression, RCC levels of _CXCL9/10/11_, _CXCR3_, _CXCL13_, _CXCR5_, _XCL1_ and _XCR5_ were found marginal as compared with age, grade, tumor and metastasis stage (__Supplementary Figure S17A__). 
As evident from a comparison of concordance indexes of the multi-parameter model consisting of solely clinical and demographic variables and the model including additionally gene expression parameters, there was no add-on value of the investigated gene expression levels to the prediction accuracy of the overall survival in the TCGA KIRC, E-MTAB 1980 or RECA-EU collectives (__Supplementary Figure S17B__).

Taken together, we could not observe differences in survival between the chemokine high and chemokine low RCC. 
None of the genes used for cluster definition correlated consistently and significantly with overall or relapse-free survival in more than two out of eight analyzed datasets. 
This suggests, that potential beneficial effects of cytotoxic T cells recruited via _CXCL9/10/11_, _CXCL13_ and _XCL1_ are likely outweighed by potent immunosuppressive properties of the RCC microenvironment.

## High *CXCL9/10/11*, *CXCL13* and *XCL1* expression is associated with interferon signaling and T cell exhaustion

To elucidate such mechanisms of activation and suppression of CD8^+^ T cell response in RCC, we delved into differences in transcriptome between chemokine high and chemokine low clusters. 
In gene set variance analysis [@Hanzelmann2013], ssGSEA scores of gene signatures of `r length(clust_reactome$common_signatures)` Reactome pathways were found significantly upregulated in chemokine high cluster cancers as compared with the chemokine low cluster in all investigated collectives. 
These included, PD-1 signaling involved in T cell exhaustion and signatures of pathways employing JAK/STAT signaling: interferon (IFN) alpha/beta, IFN gamma, interleukin 9 (IL9) and IL21 signaling (__Figure \@ref(fig:fig-clust-reactome)__, __Supplementary Table S1__). 
Of note, the upregulation of JAK/STAT along with an activated chemokine and cytokine signaling, and increased cytotoxicity in the chemokine high cluster were identified by the SPIA algorithm modelling activity of signaling pathways based on differential gene expression [@Tarca2009] (__Supplementary Figure S18__). 

To investigate the JAK/STAT signaling and T cell exhaustion processed in more details, we resorted to identification of genes differentially expressed in chemokine high cluster cancers as compared with chemokine low RCC (__Supplementary Tables S2 - S3__). Out of `r length(dge_ifn$genes)` transcripts associated with the IFN alpha/beta and IFN gamma Reactome pathways, between 
`r dge_ifn$regulation_counts %>% filter(regulation == 'upregulated', cohort != 'cm10') %>% .$n %>% min` and 
`r dge_ifn$regulation_counts %>% filter(regulation == 'upregulated', cohort != 'cm10') %>% .$n %>% max` genes were upregulated in chemokine high RCC of the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, RECA-EU, CheckMate 025 everolimus and CheckMate 025 nivolumab collectives. 
Only two IFN signaling/related genes were significantly differentially regulated in the smallest CheckMate 010 cohort. 
The top regulated IFN signaling-related genes encompassed IFN gamma (_IFNG_), genes of signaling mediators (_JAK2_, _STAT1_, _ISG15_), canonical transcriptional targets of IFN signaling (_IRF_ and _GPB_ family), anti-viral defense genes (_OAS1_, _OASL_, _SAMHB_), antibody receptors (_FCGR1B_, _TRIM21_) and genes of immunoproteasome and antigen-presenting complex (_PSMB8_, _B2M_, _HLA_ family). 
In particular, high levels of _IFNG_ and direct JAK2/STAT1 transcriptional targets suggest high activity and functionality of IFN gamma-mediated signaling in chemokine high RCC (__Supplementary Figures S19 - S20__). 
Of note, _CXCL9/10/11_ belong to classical IFN gamma-stimulated genes and as such may perpetuate cytotoxic T cell recruitment to the IFN-rich milieu of chemokine high RCC.

A sizable fraction of genes related to T cell exhaustion [@Zhang2022; @Kocher2022; @Woroniecka2018; @Cai2020] was upregulated in the chemokine high cluster as compared with chemokine low cluster RCC of the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, RECA-EU, CheckMate 025 everolimus and CheckMate 025 nivolumab collectives (total genes: `r dge_exhaust$genes %>% nrow`, upregulated genes: `r dge_exhaust$analysis_tbl[names(dge_exhaust$analysis_tbl) != 'cm10'] %>% map(filter, regulation == 'upregulated') %>% map_dbl(nrow) %>% min` - `r dge_exhaust$analysis_tbl[names(dge_exhaust$analysis_tbl) != 'cm10'] %>% map(filter, regulation == 'upregulated') %>% map_dbl(nrow) %>% max`). 
These upregulated genes included transcripts of surface co-stimulation/inhibition molecules (_TIGIT_, _CD27_, _ICOS_, _LAG3_, _TNFSRF9_), cytotoxic proteins (_GZMB_) and transcription factors driving T cell differentiation, persistence and exhaustion (_TOX_, _EOMES_, _BATF_) (__Supplementary Figures S21 - S22__). 
Genes coding for immune checkpoint proteins targeted for experimental and approved drugs, _PDCD1_, _CD274_ (PD-L1), _PDCD1LG2_ (PD-L2), _CTLA4_ and _HAVCR2_ (TIM3), were found to be expressed at significantly higher levels in the chemokine high than chemokine low cluster in the majority of analyzed cohorts (__Figure \@ref(fig:fig-checkpoint)__, __Supplementary Figure S23__). In sum, the abundant cytotoxic T cell infiltration and high activity of potentially anti-tumorigenic IFN signaling in chemokine high cluster RCC is strongly counteracted by multiple redundant immune checkpoint and T cell exhaustion processes. 

## High *CXCL9/10/11*, *CXCL13* and *XCL1* expression is paralleled by reduced oxidative energy metabolism and enhanced immunosuppressive IDO1-mediated tryptophan decay

Finally, we sought to investigate potential alterations of cellular metabolism in the chemokine high cluster as compared with chemokine low cluster RCC based on whole-genome differences in gene expression [@Gavai2015; @King2016] (__Supplementary Table S5__). 
In such analysis, substantial fractions of reactions involved in oxidative energy metabolism: fatty acid oxidation, citric acid cycle and oxidative phosphorylation were found to be inhibited in chemokine high cancers. 
In turn, one-third of reactions of tryptophan metabolism (TRP) implicated in T cell immunosuppression were significantly upregulated in chemokine high cancers of the TCGA KIRC collective (__Figure \@ref(fig:fig-meta-global)__). 
In more detail, widespread significant inhibition of multiple fatty acid oxidation reactions was observed in chemokine high cancers of the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093 and CheckMate 025 nivolumab cohorts (__Supplementary Figure S24__). 
An overall reduced activity of the citric acid cycle was particularly evident in chemokine high cluster cancers of the four largest collectives (TCGA KIRC, E-MTAB 1980, GSE73731 and GSE167093) and the 2-oxoglutarate - malate section of the cycle (__Supplementary Figure S25__). 
In oxidative phosphorylation, sustained inhibition the complexes I and IV could be detected (__Supplementary Figure S26__). 

In the chemokine high clusters of the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093 and RECA-EU cancers, significant >50% increase in activity of indoleamine 2,3-dioxygenase (IDO) was detected, a reaction responsible for TRP depletion and the first step of synthesis of immunosuppressive kynurenin and quinolinic acid. 
In the CheckMate collectives, an increased predicted IDO activity was evident, yet missed statistical significance following multiple testing correction. 
Furthermore, we could observe an elevated activity of kynureninase (KYNU), a key enzyme in quinolinate biosynthesis, in chemokine high cancers as compared with the chemokine low cluster (__Supplementary Figure S27__). 
These findings were could be corroborated at the gene expression level in the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, where both isoforms of IDO, _IDO1_ and _IDO2_ were significantly upregulated in the chemokine high cluster. 
Significantly higher levels of _IDO1_ could be detected in the RECA-EU and CheckMate 025 everolimus chemokine high cancers as well. 
_KYNU_ gene was also expressed at significantly increased levels in chemokine high cancers in four out of eight investigated cohorts (__Figure \@ref(fig:fig-trycats-genes)__, __Supplementary Figure S28__).

Results of metabolic reaction activity modeling based on differential gene expression suggest profound inhibition of oxidative energy metabolism of chemokine high cancers along with increased activity of immunosuppressive TRP - kynurenin - quinolinate pathway. 
These processes may pose another hurdles to effective anti-tumor T cell response in RCC cancers hallmarked by high expression of _CXCL9/10/11_, _CXCL13_ and _XCL1_ cytotoxic T cell-attracting chemokines.

\newpage

# Tables

```{r tab-cohort, tab.cap = 'Characteristic of the study cohorts. Numeric variables are presented as medians with interquartile ranges and ranges. Categorical variables are presented as percentage and total number within the complete observation set.'}

flextable::flextable(tables$cohort) %>% 
  flextable::width(1, width = 3, unit = 'cm') %>% 
  flextable::width(2:9, width = 2.2, unit = 'cm') %>% 
  footnote(11, 1, 
           value = as_paragraph('Complete or pertial response'),
           part = 'body', 
           ref_symbols = 'a') %>% 
  theme_vanilla

```

\newpage

# Figures

```{r fig-norm-tumor, fig.width = mm_inch(figures$norm_tumor$w), fig.height = mm_inch(figures$norm_tumor$h), fig.cap = 'Differences in expression of CXCL9/10/11, CXCL13, XCL1 and their cognate receptors between the RCC tumor tissue and normal kidney.'}

figures$norm_tumor$plot

```

__Figure \@ref(fig:fig-norm-tumor). Differences in expression of *CXCL9/10/11*, *CXCL13*, *XCL1* and their cognate receptors between the RCC tumor tissue and normal kidney.__ 

_Differences in expression of CXCL9/10/11, CXCR3, CXCL13, CXCR5, XCL1 and XCR1 between the RCC and normal kidney tissue were investigated in donor matched samples from the TCGA KIRC and GSE167093 cohorts by two-tailed paired T test. The results were corrected for multiple testing with false discovery rate (FDR) method. Points represent single samples, samples from the same tissue donors are connected with lines. P values are displayed in the plot captions. Numbers of tissue donors are indicated under the plots._

\newpage

```{r fig-xcell, fig.width = mm_inch(figures$xcell$w), fig.height = mm_inch(figures$xcell$h), fig.cap = 'Expression of CXCL9/10/11, CXCL13, XCL1 and their cognate receptors in RCC samples enriched in CD8^+^ and CD4^+^ T cells, activated myeloid DC, TAM and B cells.'}

figures$xcell$plot

```

__Figure \@ref(fig:fig-xcell). Expression of *CXCL9/10/11*, *CXCL13*, *XCL1* and their cognate receptors in RCC samples enriched in CD8^+^ and CD4^+^ T cells, activated myeloid DC, TAM and B cells.__ 

_Pairwise association of the gene expression and xCell non-immune cell infiltration estimates was assessed by scaled Spearman's correlation. The genes of interest and the cell types associated with the genes of interest with moderate-to-large strength (correlation coefficient $\rho$ > 0.4) were visualized as network force-directed plots with edge weighting by the scaled Spearman's correlation coefficient. CD8^+^ T: CD8^+^ T cells, CD4^+^ T: CD4^+^ T cells, n: naive; m: memory; em: effector memory; cm: central memory; Th1: T helper cells 1; Th1: T helper cells 1; mDC: myeloid dendritic cells, act mDC: activated mDC; pDC: plasmacytoid dendritic cells; TAM: tumor-associated macrophages; B: B cells, switch Bm: class-switched memory B cells; NK: natural killer cells; NKT: natural killer T cells; T $\gamma\delta$: $\gamma\delta$ T cells; Mono: monocytes; Eosino: eosinophils._

\newpage

```{r fig-sc-rnaseq, fig.width = mm_inch(figures$sc_rnaseq$w), fig.height = mm_inch(figures$sc_rnaseq$h), fig.cap = 'Results of scRNAseq analysis, placeholder'}

figures$sc_rnaseq$plot

```

__Figure \@ref(fig:fig-sc-rnaseq). Results of scRNAseq analysis, placeholder__ 

\newpage

```{r fig-lympho-clust, fig.width = mm_inch(figures$lympho_clust$w), fig.height = mm_inch(figures$lympho_clust$h), fig.cap = 'Expression of CXCL9/10/11, CXCL13, XCL1 and their cognate receptors in the chemokine clusters.'}

figures$lympho_clust$plot

```

__Figure \@ref(fig:fig-lympho-clust). Expression of *CXCL9/10/11*, *CXCL13*, *XCL1* and their cognate receptors in the chemokine clusters.__ 

_Chemokine clusters of RCC samples were defined in the TCGA KIRC training cohort in respect to normalized log~2~-transformed expression of CXCL9/10/11, CXCR3, CXCL13, XCL1 and XCR1 by the PAM/cosine distance algorithm. Prediction of cluster assignment in the E-MTAB 1980, GSE73731, GSE167093, RECA-EU, CheckMate 010, CheckMate 025 everolimus and CheckMate 025 nivolumab (CM 025 NIVO) was done with the inverse distance-weighted 15-nearest neighbor classifier. log~2~ transformed expression of the clustering genes was compared between the chemokine (chemox) high and low cluster by false discovery rate (FDR) corrected two-tailed T test in the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, RECA-EU collectives. Mean normalized expression values are visualized as thick lines, tinted regions represent two standard errors of the mean (SEM). P values are shown in the Y axis. Numbers of cancer samples assigned to the clusters are displayed in the plot captions._

\newpage

```{r fig-clust-infil, fig.width = mm_inch(figures$clust_infil$w), fig.height = mm_inch(figures$clust_infil$h), fig.cap = 'Predicted infiltration of T cells, activated mDC and TAM in the chemokine clusters.'}

figures$clust_infil$plot

```

__Figure \@ref(fig:fig-clust-infil). Predicted infiltration of T and B cells, activated mDC and TAM in the chemokine clusters.__ 

_Differences in levels of effector memory CD8^+^ T (CD8^+^ Tem), central memory CD8^+^ T (CD8^+^ Tcm), memory CD4^+^ T (CD4^+^ Tm), effector memory CD4^+^ T cells (CD4^+^ Tem), B cells (B), activated myeloid dendritic cells (act mDC), M1 and M2 tumor-associated macrophages (TAM) between the chemokine (chemox) high an chemokine low cluster were assessed by false discovery rate (FDR) corrected Mann-Whitney test in the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, RECA-EU collectives. Median normalized infiltration levels with interquartile ranges (IQR) are presented as boxes, whiskers span over the 150% IQR. Points represent single cancer samples. P values are indicated in the Y axes, significant effects are highlighted in bold. Numbers of cancer samples assigned to the chemokine high and low clusters are displayed in the plot captions._

\newpage

```{r fig-clust-os, fig.width = mm_inch(figures$clust_os$w), fig.height = mm_inch(figures$clust_os$h), fig.cap = 'Overall survival in the chemokine clusters.'}

figures$clust_os$plot

```

__Figure \@ref(fig:fig-clust-os). Overall survival in the chemokine clusters.__ 

_Differences in overall survival between the chemokine (chemox) high and low cluster RCC patients were investigated by false discovery rate (FDR) corrected Peto-Peto test. Fractions of surviving patients are visualized in Kaplan-Meier plots. Uncorrected and FDR-corrected p values are displayed in the plots. Number of complete observations and deaths (events) are displayed in the plot captions, numbers of patients in the clusters are indicated under the plots. CM 010: CheckMate 010; CM 025 EVER: CheckMate 025 everolimus; CM 025 NIVO: CheckMate 025 nivolumab._

\newpage

```{r fig-clust-reactome, fig.width = mm_inch(figures$clust_reactome$w), fig.height = mm_inch(figures$clust_reactome$h), fig.cap = 'Reactome pathway gene signatures significantly regulated between the chemokine clusters.'}

figures$clust_reactome$plot

```

__Figure \@ref(fig:fig-clust-reactome). Reactome pathway gene signatures significantly regulated between the chemokine clusters.__ 

_Single sample gene set enrichment analysis (ssGSEA) scores for Reactome pathway gene signatures (n = `r nrow(sig$reactome_lexicon)`) were compared between chemokine (chemox) high and chemokine low cluster cancers by false discovery rate (FDR) corrected two-tailed T test. Significance and estimated differences in ssGSEA scores for the TCGA KIRC cohort are presented in a volcano plot. Points represent single genes, point color codes for the regulation sign. The significance cutoff is visualized as a dashed line. Reactome pathway signatures found to be significantly regulated in all investigated cohorts are highlighted and labeled with their names._

\newpage

```{r fig-checkpoint, fig.width = mm_inch(figures$checkpoint$w), fig.height = mm_inch(figures$checkpoint$h), fig.cap = 'Differential expression of genes coding for clinically relevant immune checkpoint proteins in the chemokine clusters.'}

figures$checkpoint$plot

```

__Figure \@ref(fig:fig-checkpoint). Differential expression of genes coding for clinically relevant immune checkpoint proteins in the chemokine clusters.__ 

_Genes differentially expressed in chemokine (chemox) high versus chemokine low cancers were identified by false discovery rate (FDR) corrected two-tailed T test in the TCGA KIRC, E-MTAB 1980, GSE73731, GSE167093, RECA-EU collectives. Results for clinically relevant immune checkpoint genes are presented. Median normalized log~2~ expression levels with interquartile ranges (IQR) are presented as boxes, whiskers span over the 150% IQR. Points represent single cancer samples. P values are indicated in the Y axes, significant effects are highlighted in bold. Numbers of cancer samples assigned to the chemokine high and low clusters are displayed in the plot captions._

\newpage

```{r fig-meta-global, fig.width = mm_inch(figures$meta_global$w), fig.height = mm_inch(figures$meta_global$h), fig.cap = 'Differential regulation of metabolic reactions between the chemokine clusters.'}

figures$meta_global$plot

```

__Figure \@ref(fig:fig-meta-global). Differential regulation of metabolic reactions between the chemokine clusters.__ 

_Regulation of the Recon human metabolism model reactions in chemokine (chemox) high versus chemokine low cluster cancers was predicted based on the differential gene expression with the BiGGR and biggrExtra package tools. Significance of the metabolic reaction regulation was assessed by Monte Carlo simulation and corrected for multiple testing with the false discovery rate (FDR) method. Fractions of significantly activated (A) and significantly inhibited (B) reactions within the Recon model subsystems are presented for the TCGA KIRC cohort. Fatty acid oxidation, citric acid cycle, oxidative phosphorylation and tryptophan (TRP) metabolism subsystems investigated in more detail are highlighted in bold._

\newpage

```{r fig-trycats-genes, fig.width = mm_inch(figures$trycats_genes$w), fig.height = mm_inch(figures$trycats_genes$h), fig.cap = 'Differential expression of genes involved in the tryptophan/kynurenine/quinolinate metabolism in the chemokine clusters.'}

figures$trycats_genes$plot

```

__Figure \@ref(fig:fig-trycats-genes). Differential expression of genes involved in the tryptophan/kynurenine/quinolinate metabolism in the chemokine clusters.__ 

_Genes differentially expressed in chemokine (chemox) high versus chemokine low cancers were identified by false discovery rate (FDR) corrected two-tailed T test. Results for genes involved in the tryptophan - kynurenin - quinolinate metabolic pathway are presented._

_(A) Pathway scheme. TRP: tryptophan; TDO: tryptophan 2,3-dioxygenase; KMO: kynurenine 3-monooxygenase; KYNU: kynureninase; KAT: kynurenin aminotransferase; HAAO: 3-hydroxyanthranilate 3,4-dioxygenase._

_(B) Gene expression in the clusters. Median normalized log~2~ expression levels with interquartile ranges (IQR) are presented as boxes, whiskers span over the 150% IQR. Points represent single cancer samples. P values are indicated in the Y axes, significant effects are highlighted in bold. Numbers of cancer samples assigned to the chemokine high and low clusters are displayed in the plot captions. IDO: indoleamine 2,3-dioxygenase; KYAT1: kynurenine—oxoglutarate transaminase 1. _ 

\newpage

```{r fig-summary, fig.width = mm_inch(figures$summary$w), fig.height = mm_inch(figures$summary$h), fig.cap = 'Recruitment of CD8^+^ T cells via CXCL9/10/11, CXCL13 and XCL1 and suppression of CD8^+^ T cells in the microenvironment of RCC.'}

figures$summary$plot

```

__Figure \@ref(fig:fig-summary). Recruitment of CD8^+^ T cells via *CXCL9/10/11*, *CXCL13* and *XCL1*, and suppression of CD8^+^ T cells in the microenvironment of RCC.__ 

_(A) Cellular sources of *CXCL9/10/11*, *CXCL13* and *XCL1* in the RCC microenvironment._

_(B) Proposed negative feedback between the *CXCL9/10/11*-, *CXCL13*- and *XCL1*-mediated cytotoxic T recruitment, interferon (IFN) - JAK/STAT signaling, T cell suppression and exhaustion by immune checkpoint proteins, suppression of oxidative energy metabolism and tryptophan decay to immunosuppressive metabolites._

\newpage

# References
